util.go

```diff
@@ -0,0 +1,80 @@
+package files
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"sort"
+)
+
+// ToFile is an alias for n.(File). If the file isn't a regular file, nil value
+// will be returned
+func ToFile(n Node) File {
+	f, _ := n.(File)
+	return f
+}
+
+// ToDir is an alias for n.(Directory). If the file isn't directory, a nil value
+// will be returned
+func ToDir(n Node) Directory {
+	d, _ := n.(Directory)
+	return d
+}
+
+// FileFrom is a convenience function which tries to extract or create new file
+// from provided value. If a passed value can't be turned into a File, nil will
+// be returned.
+//
+// Supported types:
+// * files.File (cast from Node)
+// * DirEntry / DirIterator (cast from e.Node())
+// * []byte (wrapped into NewReaderFile)
+// * io.Reader / io.ReadCloser (wrapped into NewReaderFile)
+func FileFrom(n interface{}) File {
```

Comment:
We could (mostly) replace this with:

1. A modified `NewReaderFile` that takes a `io.Reader` and auto-upgrades with NopCloser if necessary.
2. A `NewBytesFile` constructor.